### PR TITLE
sql: add missing dynamic privileges and update optm lock version

### PR DESCRIPTION
**Pull request by shichun-0415**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
This PR fixes two issues:
- issue 1: https://github.com/pingcap/docs-cn/issues/11670. Reference: https://docs.pingcap.com/zh/tidb/stable/optimistic-transaction#:~:text=%E8%87%AA%20v3.0.8%20%E5%BC%80%E5%A7%8B%EF%BC%8CTiDB%20%E9%9B%86%E7%BE%A4%E9%BB%98%E8%AE%A4%E4%BD%BF%E7%94%A8%E6%82%B2%E8%A7%82%E4%BA%8B%E5%8A%A1%E6%A8%A1%E5%BC%8F%E3%80%82
- issue 2: six dynamic privileges are missing in the user doc and should be documented

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.6 (TiDB 6.6 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [x] v6.4 (TiDB 6.4 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: privilege-management.md [hunk 1]
```diff
@@ -261,13 +261,19 @@ SHOW GRANTS FOR `rw_user`@`192.168.%`;
...
 * `BACKUP_ADMIN`
 * `RESTORE_ADMIN`
+* `SYSTEM_USER`
+* `SYSTEM_VARIABLES_ADMIN`
 * `ROLE_ADMIN`
-* `PLACEMENT_ADMIN` 允许创建、删除和修改放置策略（placement policy）。
 * `CONNECTION_ADMIN`
-* `SYSTEM_VARIABLES_ADMIN`
+* `PLACEMENT_ADMIN` 允许创建、删除和修改放置策略 (placement policy)。
+* `DASHBOARD_CLIENT` 允许登录 TiDB-Dashboard。
```
Comment (Oreoxmt) [inline privilege-management.md:271]:
```suggestion
* `DASHBOARD_CLIENT` 允许登录 TiDB Dashboard。
```

File: privilege-management.md [hunk 2]
```diff
@@ -261,13 +261,19 @@ SHOW GRANTS FOR `rw_user`@`192.168.%`;
...
+* `RESTRICTED_TABLES_ADMIN` 允许在 SEM 打开的情况下查看系统表。
+* `RESTRICTED_STATUS_ADMIN` 允许在 SEM 打开的情况下查看 [`SHOW [GLOBAL|SESSION] STATUS`](/sql-statements/sql-statement-show-status.md) 中的状态变量。
+* `RESTRICTED_VARIABLES_ADMIN` 允许在 SEM 打开的情况下查看所有系统变量。
+* `RESTRICTED_USER_ADMIN` 不允许在 SEM 打开的情况下使用 `SUPER` 用户撤销访问权限。
+* `RESTRICTED_CONNECTION_ADMIN` 允许 kill 属于 `RESTRICTED_USER_ADMIN` 用户的连接。该权限对 `KILL` 和 `KILL TIDB` 语句生效。
```
Comment (Oreoxmt) [inline privilege-management.md:276]:
```suggestion
* `RESTRICTED_CONNECTION_ADMIN` 允许 KILL 属于 `RESTRICTED_USER_ADMIN` 用户的连接。该权限对 `KILL` 和 `KILL TIDB` 语句生效。
```

**Review by Oreoxmt — APPROVED** [review]
Rest LGTM

**Review by djshow832 — APPROVED** [review]

**Pull request merged**